app/helpers/page_helper.rb

```diff
@@ -0,0 +1,18 @@
+module PageHelper
+
+  # include so append_white_list_params is available during rspec tests as the controller helpers are not automatically loaded
+  include ApplicationHelper
+
+  def redirect_to_page_with_run_path(sequence, activity_id, page_id, run_key, query_parameters = {})
+    redirect_to append_white_list_params get_page_with_run_path(sequence, activity_id, page_id, run_key, query_parameters)
+  end
+
+  def get_page_with_run_path(sequence, activity_id, page_id, run_key, query_parameters = {})
+    if sequence
+      sequence_page_with_run_path(sequence, activity_id, page_id, run_key, query_parameters)
+    else
+      page_with_run_path(activity_id, page_id, run_key, query_parameters)
```

Comment:
Since this is a public helper method we should put the `append_white_list_params` on these calls instead in the `redirect_to_...` method.